.github/actions/artifact_download/action.yml

```diff
@@ -0,0 +1,39 @@
+name: Download artifact
+description: Download and decrypt an artifact.
+
+inputs:
+  name:
+    description: 'The name of the artifact.'
+    required: true
+  path:
+    description: 'Download to a specified path.'
+    required: false
+    default: ./
+  encryption-secret:
+    description: 'The secret to use for decrypting the artifact.'
+    required: true
```

Comment:
Maybe it can be an optional input? This way we get the best of both worlds.

Reply:
In my experience this doesn't really work since I believe that composite actions are prevented from reading secrets. I originally tried to read it directly in the action and it didn't work, I remember the docs telling me the reason was the one above.

Reply:
Not the official docs, but [here is a stackoverflow post with the same problem](https://stackoverflow.com/questions/70098241/using-secrets-in-composite-actions-github)